### ISSUE TITLE
error: too many arguments to function ‘getDirFromMousePos’

### ISSUE THREAD

**Issue by KuriKai** [acoto87/war1#7]
/build-gcc-linux64.sh 
In file included from ../../src/war1.c:139:
../../src/war_map.c: In function ‘updateMapCursor’:
../../src/war_map.c:1903:24: error: too many arguments to function ‘getDirFromMousePos’
 1903 |             vec2 dir = getDirFromMousePos(context, input);
      |                        ^~~~~~~~~~~~~~~~~~
In file included from ../../src/war1.c:81:
../../src/war.h:87:6: note: declared here
   87 | vec2 getDirFromMousePos(WarContext* context)
      |      ^~~~~~~~~~~~~~~~~~
In file included from ../../src/war1.c:147:
../../src/war_ai.c: In function ‘createAI’:
../../src/war_ai.c:6:26: warning: passing argument 1 of ‘WarAICommandListInit’ from incompatible pointer type [-Wincompatible-pointer-types]
    6 |     WarAICommandListInit(&ai->currentCommands, WarAICommandListDefaultOptions);
      |                          ^~~~~~~~~~~~~~~~~~~~
      |                          |
      |                          WarAICommandQueue * {aka struct <anonymous> *}
In file included from ../../src/war1.c:64:
../../deps/include/shl/list.h:114:37: note: expected ‘WarAICommandList *’ {aka ‘struct <anonymous> *’} but argument is of type ‘WarAICommandQueue *’ {aka ‘struct <anonymous> *’}
  114 |     void typeName ## Init(typeName* list, typeName ## Options options) \
../../src/war_types.h:1977:1: note: in expansion of macro ‘shlDefineList’
 1977 | shlDefineList(WarAICommandList, WarAICommand*)
      | ^~~~~~~~~~~~~
In file included from ../../src/war1.c:147:
../../src/war_ai.c:7:27: warning: passing argument 1 of ‘WarAICommandQueueInit’ from incompatible pointer type [-Wincompatible-pointer-types]
    7 |     WarAICommandQueueInit(&ai->nextCommands, WarAICommandQueueDefaultOptions);
      |                           ^~~~~~~~~~~~~~~~~
      |                           |
      |                           WarAICommandList * {aka struct <anonymous> *}
In file included from ../../src/war1.c:65:
../../deps/include/shl/queue.h:87:37: note: expected ‘WarAICommandQueue *’ {aka ‘struct <anonymous> *’} but argument is of type ‘WarAICommandList *’ {aka ‘struct <anonymous> *’}
   87 |     void typeName ## Init(typeName *queue, typeName ## Options options) \
../../src/war_types.h:1974:1: note: in expansion of macro ‘shlDefineQueue’
 1974 | shlDefineQueue(WarAICommandQueue, WarAICommand*)
      | ^~~~~~~~~~~~~~
In file included from ../../src/war1.c:147:
../../src/war_ai.c: In function ‘createAICommand’:
../../src/war_ai.c:22:5: warning: implicit declaration of function ‘WarAIRequestQueuePush’; did you mean ‘WarAICommandQueuePush’? [-Wimplicit-function-declaration]
   22 |     WarAIRequestQueuePush(&ai->nextCommands, command);
      |     ^~~~~~~~~~~~~~~~~~~~~
      |     WarAICommandQueuePush
../../src/war_ai.c: In function ‘processAICurrentCommands’:
../../src/war_ai.c:209:34: warning: initialization of ‘WarAICommandList *’ {aka ‘struct <anonymous> *’} from incompatible pointer type ‘WarAICommandQueue *’ {aka ‘struct <anonymous> *’} [-Wincompatible-pointer-types]
  209 |     WarAICommandList* commands = &ai->currentCommands;
      |                                  ^
../../src/war_ai.c: In function ‘processAINextCommands’:
../../src/war_ai.c:229:41: warning: initialization of ‘WarAICommandList *’ {aka ‘struct <anonymous> *’} from incompatible pointer type ‘WarAICommandQueue *’ {aka ‘struct <anonymous> *’} [-Wincompatible-pointer-types]
  229 |     WarAICommandList* currentCommands = &ai->currentCommands;
      |                                         ^
../../src/war_ai.c:230:39: warning: initialization of ‘WarAICommandQueue *’ {aka ‘struct <anonymous> *’} from incompatible pointer type ‘WarAICommandList *’ {aka ‘struct <anonymous> *’} [-Wincompatible-pointer-types]
  230 |     WarAICommandQueue* nextCommands = &ai->nextCommands;
      |                                       ^
../../src/war_ai.c:227:10: warning: unused variable ‘processNext’ [-Wunused-variable]
  227 |     bool processNext = true;
      |          ^~~~~~~~~~~

**Comment by acoto87** [comment]
Thanks for the report! 

I know things may be broken now, since I'm implementing new stuff into the game and I've been uploading the new code to master directly. I think is time to create a new branch for this kind of development to leave master as a stable branch. Gonna setup that tomorrow.

I'm gonna let you know when it's fixed.

**Comment by acoto87** [comment]
I uploaded some changes on `master` that fixes the compilation issues.

I also created a new branch `dev`, in which all development is going to be made. This branch can be broken as I develop the features of the game. I'm gonna keep `master` as the stable branch, in which things should not be broken. Once I finished with a feature on `dev` I'm going to merge it to `master`.

I'm closing this issue now.
